### ISSUE TITLE
LaTeX Error: File `support/latex/sbabook/sbabook.cls' not found

### ISSUE THREAD

**Issue by dmitrizagidulin** [SquareBracketAssociates/EnterprisePharo#55]
I can't get the PDF to build correctly for some reason. (Same thing happens when I try to build https://github.com/SquareBracketAssociates/DynamicWebDevelopmentWithSeaside, actually).

I ran `download.sh` - no prob.
Ran `./pillar export`.
Then `bash pillarPostExport.sh`.
Got the following error.

```
This is LuaTeX, Version beta-0.79.1 (TeX Live 2014) (rev 4971)  (format=lualatex 2014.5.25)  20 OCT 2015 17:30
 restricted \write18 enabled.
 file:line:error style messages enabled.
**Teapot.tex
(./Teapot.tex
LaTeX2e <2014/05/01>
Babel <3.9k> and hyphenation patterns for 78 languages loaded.

! LaTeX Error: File `root/support/latex/sbabook/sbabook.cls' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: cls)

Enter file name:
./Teapot.tex:5: Emergency stop.
<read *>

l.5   \usepackage
               {import}^M

Here is how much of LuaTeX's memory you used:
 13 strings out of 494695
 100000,89155 words of node,token memory allocated
 190 words of node memory still in use:
   31 glue_spec, 1 if_stack, 1 dir nodes
   avail lists: 2:11,3:1
 3849 multiletter control sequences out of 65536+600000
 14 fonts using 562767 bytes
 10i,0n,7p,85b,8s stack positions out of 5000i,500n,10000p,200000b,100000s
./Teapot.tex:5:  ==> Fatal error occurred, no output PDF file produced!
```
Please advise? :)

**Comment by cdlm** [comment]
`sbabook.cls` is in a submodule; try `git submodule update`.

There is a line for that in `compile.sh` but you skipped it by running pillar directly.

**Comment by dmitrizagidulin** [comment]
@cdlm Thanks!
(I only mentioned the running the pillar directly part since the instructions implied it was equivalent to running `compile.sh`.)
Running `./compile.sh` gave me a different error:

```
! LaTeX Error: File `datetime2.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name:
./root/support/latex/common.tex:10: Emergency stop.
<read *>

l.10


Here is how much of LuaTeX's memory you used:
 30105 strings out of 494695
 100000,794899 words of node,token memory allocated
 666 words of node memory still in use:
   2 hlist, 1 rule, 1 kern, 2 attribute, 138 glue_spec, 2 attribute_list, 1 if_s
tack, 2 write, 1 dir, 1 pdf_colorstack nodes
   avail lists: 2:17,3:1,4:7,6:6,7:1,9:2
 33037 multiletter control sequences out of 65536+600000
 33 fonts using 2596087 bytes
 67i,1n,90p,818b,220s stack positions out of 5000i,500n,10000p,200000b,100000s
./root/support/latex/common.tex:10:  ==> Fatal error occurred, no output PDF fil
e produced!
Can't generate the PDF!
```

**Comment by cdlm** [comment]
Either the package is not in your TeXlive install (try `tlmgr install datetime2`) or you're using an old release (we use TeXlive 2015)

**Comment by dmitrizagidulin** [comment]
Ah, I see. I didn't know about TeXlive. I installed LuaTeX from here http://www.luatex.org/download.html 

```
$ which luatex
/Users/dmitriz/util/context/tex/texmf-osx-64/bin/luatex
$ luatex
This is LuaTeX, Version beta-0.80.0 (TeX Live 2015) (rev 5238)
 \write18 enabled.
```

Oh, hmm, it does seem to be TeX Live 2015.. maybe the old PDFlatex is also in the path still somewhere... Will check.

**Comment by dmitrizagidulin** [comment]
Ok, I removed the older pdflatex dirs from the path, and now it gives me:

```
pillarPostExport.sh: line 9: lualatex: command not found
```
So I clearly don't have the right texlive distribution. Can you please point me to the correct one? There's nothing about this in the pillar book install instructions.

**Comment by cdlm** [comment]
You probably got a very minimal TeXlive, or even just the luatex package of a TeXlive. In which case, I would be surprised anything works at all. I usually just install the complete TeXlive, even though it's huge.

**Comment by cdlm** [comment]
http://texlive.org

There is just one TeXlive distribution (well, one release each year). The difference is how much of it do you install. If you're on Mac, there is http://www.tug.org/mactex/index.html (it adds a few mac apps to a normal TeXlive, but I never really used it). If you're on linux, your distro should have texlive packages.

Otherwise, my preferred installation path is the netinstall (download a small installer script, run/pick options, then wait for the rest to download and install)
http://tug.org/texlive/acquire-netinstall.html

**Comment by dmitrizagidulin** [comment]
Ok, I think I understand now. I had MacTex 2014 installed previously. I thought luatex was a different distribution, hence trying to install those packages directly.

I'm updating to mactex 2015 now (the live update utility didn't work, so I'm uninstalling the old & reinstalling the new version). Hopefully that'll sort it out.

Thanks again for your help.

**Comment by dmitrizagidulin** [comment]
Ok, getting closer. (hopefully). I've installed TeX Live 2015 (via maxtex). Added /usr/texbin to my path.
Now the error is:

```
Compiling PDF from book-result/Teapot/Teapot.tex

kpathsea: Running mktexfmt lualatex.fmt
/usr/texbin/mktexfmt: kpsewhich -var-value=TEXMFROOT failed, aborting early.
BEGIN failed--compilation aborted at /usr/texbin/mktexfmt line 25.
cat: Teapot.log: No such file or directory
Can't generate the PDF!
```

**Comment by dmitrizagidulin** [comment]
I think I've got it. (The error above was being caused by - it was using the wrong `kpsewhich` from an older latex install.)
Thanks again for your help, I'm closing the issue.

**Comment by dmitrizagidulin** [comment]
(I updated the installation and book generation instructions in the readme).

**Comment by cdlm** [comment]
You can mention the numbers of related issues in commit messages, and conversely; for instance, in commit a3692fe, writing `closes #55` would have added a link here and closed the issue at the same time. See https://help.github.com/articles/closing-issues-via-commit-messages/
